packages/orbo/src/index.tsx

```diff
@@ -140,23 +140,29 @@ export function createGlobalState<T>(config: GlobalStateConfig<T>) {
   ): SubContext<T> {
     let subContext = globalStateContext.subContexts.get(stateKey);
     if (!subContext) {
-      let value = config.initialState(globalStateContext.initialValues);
-      // Update state has the same shape like React's setState
-      // an can be alled in onSubscribe or by the global state setter hook
-      const updateState = (newState: T | ((prev: T) => T)) => {
-        value =
-          typeof newState === "function"
-            ? (newState as (prev: T) => T)(value)
-            : newState;
-        listeners.forEach((setter) => setter(value));
-      };
       const listeners = new Set<(newState: any) => any>();
       const newSubContext: SubContext<T> = {
         initialized: true,
-        value,
+        value: config.initialState(globalStateContext.initialValues),
         listeners,
-        updateState,
-        cleanup: onSubscribe(updateState, value),
+        updateState:
+          // Update state has the same shape like React's setState
+          // an can be alled in onSubscribe or by the global state setter hook
+          (newState: T | ((prev: T) => T)) => {
+            newSubContext.value =
+              typeof newState === "function"
+                ? (newState as (prev: T) => T)(newSubContext.value)
+                : newState;
+            listeners.forEach((setter) => setter(newSubContext.value));
+
+            // Update the stored value of the corresponding subContext
+            // This is important for newly mounted components using existing and initialized subContexts
+            const oldSubContext = globalStateContext?.subContexts.get(stateKey);
+            if (oldSubContext) {
+              oldSubContext.value = newSubContext.value;
+            }
+          },
```

Comment:
This doesn't make any sense to me - can you explain how this could ever have any effect?

```suggestion
            listeners.forEach((setter) => setter(newSubContext.value));
```

Reply:
You're right. With moving the initialization around, we don't need this code.